src/ucp/proto/proto_multi.inl

```diff
@@ -66,8 +66,11 @@ ucp_proto_multi_max_payload(ucp_request_t *req,
     size_t max_frag = lpriv->max_frag - hdr_size;
     size_t max_payload;
 
-    ucs_assertv(lpriv->max_frag > hdr_size, "max_frag=%zu hdr_size=%zu",
-                lpriv->max_frag, hdr_size);
+    /* If header is greater than max_frag - that's ok, we allow it, but we send
+     * only the header data and keep the payload empty */
+    if (lpriv->max_frag <= hdr_size) {
```

Comment:
maybe smth like
```
ssize_t max_frag;
...
max_frag = lpriv->max_frag - hdr_size;
if (max_frag < 0) {
     return 0;
}
```

Reply:
ok, good idea

Reply:
AFAIR `lpriv->max_frag` can be set to `lane_perf->max_frag` which is a transport limitation, so may there be a situation when some lanes don't really support such big header size so we cannot send even header?

Reply:
As I see in `ucp_worker_set_max_am_header` the `worker->max_am_header` is set so that header can fit in any lane. If user sends something larger than that, then we reject this input:
```C
ucp_am_send_nbx_check_header_length(ucp_worker_h worker, size_t header_length)
{
    if (ENABLE_PARAMS_CHECK && (header_length > worker->max_am_header)) {
```

Reply:
in this case, better not calculate `max_frag = lpriv->max_frag - hdr_size` during variable declaration, but do it after checking `(lpriv->max_frag <= hdr_size)`